python/lsst/ts/MTMount/mock/controller.py

```diff
@@ -79,7 +107,61 @@ def __init__(self, reply_port, log):
         self.command_dict[enums.CommandCode.BOTH_AXES_TRACK] = self.do_both_axes_track
 
         self.read_loop_task = asyncio.Future()
-        self.start_task = asyncio.create_task(self.start())
+        self.telemetry_task = asyncio.Future()
+        self.start_task = asyncio.create_task(self.connect())
+
+    async def telemetry_loop(self):
+        # For now set fake data in all fields so the telemetry topics have
+        # semi-realistic lengths; if we decide that the MCS will publish
+        # telemetry then obviously the data will have to be properly set
+        # to match the state of the mock devices.
+        print("telemetry_loop: setting arbitrary telemetry values")
+        for tel_name, data_dict in self.telemetry_dict.items():
+            for name, data in data_dict.items():
+                if type(data) is str:
+                    data = "idle"
+                elif type(data) is float:
+                    data = 123.456789
+                elif type(data) is int:
+                    data = 123456789
+                elif type(data) is bool:
+                    data = True
+                else:
+                    raise RuntimeError(
+                        f"Unsupported field type for {tel_name}.{name}: "
+                        f"{type(data).__name__}"
+                    )
+                data_dict[name] = data
+
+        try:
+            t0margin = time.monotonic()
+            while True:
+                t0 = time.monotonic()
+                for name, data_dict in self.telemetry_dict.items():
+                    # data = json.dumps(data_dict)
+                    # reply = replies.TelemetryReply(name=name, data=data)
+                    # await self.communicator.write(reply)
+                    reply = replies.TelemetryOSS5Reply(**self.telemetry_dict["OSS5"])
+                    await self.communicator.write(reply)
+                t1 = time.monotonic()
+                margin = self.telemetry_interval - (t1 - t0)
+                self.telemetry_margins.append(margin)
+                if margin > 0:
+                    await asyncio.sleep(margin)
+                if t1 - t0margin > 10:
+                    dt = t1 - t0margin
+                    t0margin = t1
+                    margin_arr = np.array(self.telemetry_margins)
+                    print(
+                        f"margin min={margin_arr.min():0.3f}; "
+                        f"max={margin_arr.max():0.3f}; "
+                        f"mean={margin_arr.mean():0.3f}; "
+                        f"std={margin_arr.std():0.3f}; "
+                        f"rate={len(margin_arr)/dt:0.1f}"
+                    )
+                    self.telemetry_margins = []
+        except Exception:
+            self.log.exception("telemetry_loop failed")
```

Comment:
may be useful to log the traceback.

Reply:
That is exactly what `log.exception` does. Note that the exception *should not* be provided to `log.exception` (an common mistake and one I just noticed today in ts_ataos).